Use "ghostty" subdirectory for resources

gpanders opened the pull request:
Installing resources directly under `${prefix}/share` causes conflicts with other packages. This will become more problematic whenever Ghostty is opened and becomes packaged in distributions.

Instead, install all resources under a "ghostty" subdirectory (i.e. `${prefix}/share/ghostty`). This includes themes, shell integration, and terminfo files.

Only `/usr/share` style paths use the "ghostty" subdirectory. On macOS, Ghostty is already isolated within its app bundle, and if `$GHOSTTY_RESOURCES_DIR` is set then we assume that points to the actual resources dir (without needing to append "ghostty" to it).

This PR also adds a build time `-Dresources-dir` option. This was suggested by @jcollie in Discord as something that might be useful for packagers. If omitted (the default), then Ghostty's behavior is the same as it is today (it traverses the filesystem upward relative to the `ghostty` binary looking for the terminfo file).

gpanders committed:
core: look for resources in "ghostty" subdirectory for "share" paths

Installing resources directly under ${prefix}/share causes conflicts
with other packages. This will become more problematic whenever Ghostty
is opened and becomes packaged in distributions.

Instead, install all resources under a "ghostty" subdirectory (i.e.
${prefix}/share/ghostty). This includes themes, shell integration, and
terminfo files.

Only "/usr/share" style paths use the "ghostty" subdirectory. On macOS,
Ghostty is already isolated within its app bundle, and if
$GHOSTTY_RESOURCES_DIR is set then we assume that points to the actual
resources dir (without needing to append "ghostty" to it).

mitchellh reviewed (commented):
One comment that probably complicates this PR a bit. It's the main reason I didn't do this earlier! But, your work looks good and if you want you can consider that one complication its not _too bad_.

gpanders committed:
macos: include ghostty subdirectory under Resources

This enables us to treat Ghostty's resources directory the same way
whether it is installed in a macOS app bundle or under e.g. /usr/share.

mitchellh committed:
macos: reordering some files
